src/techui_builder/generate.py

```diff
@@ -157,6 +214,68 @@ def _create_widget(self, component: Entry) -> EmbeddedDisplay | ActionButton:
 
         return new_widget
 
+    def layout_widgets(self, widgets: list[EmbeddedDisplay | ActionButton]):
+        group_spacing = 30
+        max_group_height = 800
+        spacing_x = 20
+        spacing_y = 30
+        # Group tiles by size
+        groups = defaultdict(list)
+        for widget in widgets:
+            key = self._get_screen_dimensions_object(widget)
+
+            groups[key].append(widget)
+
+        # Sort groups by width (optional)
+        sorted_widgets = []
+        sorted_groups = sorted(groups.items(), key=lambda g: g[0][0], reverse=True)
+        current_x = 0
+        current_y = 0
+        column_width = 0
+        column_levels = []
+
+        for (h, w), group in sorted_groups:
+            for widget in group:
+                placed = False
+                for level in column_levels:
+                    level_y, level_x = self._get_screen_position_object(level[0])
+                    widget_height, widget_width = self._get_screen_dimensions_object(
+                        widget
+                    )
+                    level_width = (
+                        sum(
+                            (self._get_screen_dimensions_object(t))[1] + spacing_x
+                            for t in level
+                        )
+                        - spacing_x
+                    )
+                    if (
+                        level_y + h <= max_group_height
+                        and level_width + widget_width <= column_width
+                    ):
+                        hh, width_1 = self._get_screen_dimensions_object(level[-1])
+                        y_1, x_1 = self._get_screen_position_object(level[-1])
```

Comment:
`hh` and `y_1` are not used. Are they needed? If not, just call them `_`
```suggestion
                        _, width_1 = self._get_screen_dimensions_object(level[-1])
                        _, x_1 = self._get_screen_position_object(level[-1])
```